ui/apps/platform/src/types/reportConfigurationService.proto.ts

```diff
@@ -0,0 +1,85 @@
+import { VulnerabilitySeverity } from './cve.proto';
+
+export type ReportType = 'VULNERABILITY';
+
+export type ReportConfiguration = {
+    id: string;
+    name: string;
+    description: string;
+    type: ReportType;
+    vulnReportFilters: VulnerabilityReportFilters;
+    emailConfig: EmailNotifierConfiguration;
+    schedule: Schedule;
+    resourceScope: ResourceScope;
+};
+
+type VulnerabilityReportFiltersBase = {
+    fixability: Fixability;
+    severities: VulnerabilitySeverity[];
+    imageTypes: ImageType[];
+};
+
+export type VulnerabilityReportFilters =
+    | (VulnerabilityReportFiltersBase & {
+          allVuln: boolean;
+      })
+    | (VulnerabilityReportFiltersBase & {
+          lastSuccessfulReport: boolean;
+      })
+    | (VulnerabilityReportFiltersBase & {
+          startDate: string; // in the format of google.protobuf.Timestamp};
+      });
+
+export type Fixability = 'BOTH' | 'FIXABLE' | 'NOT_FIXABLE';
+
+export type ImageType = 'DEPLOYED' | 'WATCHED';
+
+export type EmailNotifierConfiguration = {
+    notifierId: string;
+    mailingLists: string[];
+};
+
+type ScheduleBase = {
+    intervalType: IntervalType;
+    hour: number;
+    minute: number;
+};
+
+export type Schedule =
+    | (ScheduleBase & {
+          weekly: WeeklyInterval;
+      })
+    | (ScheduleBase & {
+          daysOfWeek: DaysOfWeek;
+      })
+    | (ScheduleBase & {
+          daysOfMonth: DaysOfMonth;
+      });
+
+export type IntervalType = 'UNSET' | 'DAILY' | 'WEEKLY' | 'MONTHLY';
+
+export type Interval = DaysOfWeek | DaysOfMonth;
+
+// TODO - Note that the types of `days` below are not exact, the API returns `number[]`, but the UI converts
```

Comment:
Nit: If the actual value returned from the server is of type `number`, can we declare that in the type and make an explicit conversion in the UI if we really need it to be `string`?

Reply:
Sounds good to me. I copy-pasted these scheduling types from the old reporting workflow. We can make sure to keep them the same as what the API returns. Good catch